app/controllers/pdf_converter_controller.rb

```diff
@@ -1,48 +1,65 @@
 require 'pdf/reader'
 require 'wordpress_uitlity'
 require 'rmagick'
+require 'zip'
 
 class PdfConverterController < ApplicationController
-  before_filter :wordpress_utility, only: :publish
+  before_filter :wordpress_utility, :specify_file, only: :publish
   after_filter :cleanup_temp_files, only: :publish
 
   def new
   end
 
   def publish
-    @link = @wordpress_utility.publish(*extract_images_and_text)
+    @link = @wordpress_utility.publish(*[@content, @images.flatten])
```

Comment:
instead of array use return value of `extract_blog_content` and pass it to utilities `publish` method